### PR TITLE
Sphinx cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**
No real changes, just using modern types and extracting hmac into another file for reuse.

Changelog-None

### CODE REVIEW

File: common/hmac.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#include <bitcoin/privkey.h>
+#include <ccan/array_size/array_size.h>
+#include <ccan/mem/mem.h>
+#include <common/hmac.h>
+#include <wire/wire.h>
+
+void hmac(const void *src, size_t slen,
+	  const void *key, size_t klen,
```
Comment (cdecker) [inline common/hmac.c]: Seeing the previous cleanup would we ever want to use anything other than a `struct secret`?

File: common/sphinx.c [hunk 2]
```diff
@@ -198,43 +197,29 @@ static void xor_cipher_stream(void *dst, const struct secret *k, size_t dstlen)
 	crypto_stream_chacha20_xor(dst, dst, dstlen, nonce, k->data);
 }
 
-static bool compute_hmac(
-	void *dst,
-	const void *src,
-	size_t len,
-	const void *key,
-	size_t keylen)
+static void compute_hmac(const u8 *src, size_t slen,
```
Comment (cdecker) [inline common/sphinx.c]: This wrapper doesn't do much, maybe we can remove it as well?

File: common/hmac.c (hunk 1 above)
Comment (rustyrussell) [reply common/hmac.c]: Sigh, unfortunately the hmac construction used by Sphinx uses the string as the *key* not src. :(

Found this out when I tried cleaning up the first time.

File: common/sphinx.c (hunk 2 above)
Comment (rustyrussell) [reply common/sphinx.c]: Will do in another PR.

Comment (rustyrussell) [reply common/sphinx.c]: Actually, it is kinda useful since it takes a secret.

**Pull request merged**